### PR TITLE
Add mkswifisdcard plugin

### PR DESCRIPTION
**Pull request by om2804**

### CODE REVIEW

**Review by jneilliii — CHANGES_REQUESTED** [review]
Quick question @om2804, where is `printer` defined in your plugin? I see multiple references to it so a little confused how that relates to `self._printer` in the scope of the plugin. 

Only other real issue I see on first look is that you are missing a [software update hook](https://docs.octoprint.org/en/master/bundledplugins/softwareupdate.html#octoprint-plugin-softwareupdate-check-config) to allow users to get notified of new releases as they are made available on your repository. Which also means you need to make sure with new versions that you create a new release with a tag that matches the plugin_version defined in setup.py. So you will also need one for 1.0.0 as first release.

**Comment by om2804** [comment]
> Quick question @om2804, where is `printer` defined in your plugin? I see multiple references to it so a little confused how that relates to `self._printer` in the scope of the plugin.

printer send to handler hook "custom_upload_to_sd" as argument

**Comment by cp2004** [comment]
You always need to specify a timeout on network requests to avoid any problems with it hanging infinitely (https://github.com/om2804/octoprint_mkswifisdcard/blob/main/octoprint_mkswifisdcard/__init__.py#L72).

Some error handling and reporting problems to the user could be required as well. Network requests are not always guaranteed to succeed.

**Comment by jneilliii** [comment]
> printer sent to handler hook "custom_upload_to_sd" as argument

ah, totally missed that. my apologies. once these other changes have been completed we can move forward with merging this. Please make sure to address Charlie's concerns as well relative to request timeout.

**Comment by om2804** [comment]
@cp2004 @jneilliii 
I made changes in all comments

File: _plugins/mkswifisdcard.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+description: Octoprint plugin for fast upload file to SD-card over MKS WiFi module
+authors:
+- om2804
+license: AGPLv3
+
+date: 2022-07-31
+
+homepage: https://github.com/om2804/octoprint_mkswifisdcard
+source: https://github.com/om2804/octoprint_mkswifisdcard
+archive: https://github.com/om2804/octoprint_mkswifisdcard/archive/master.zip
```
Comment (jneilliii) [inline _plugins/mkswifisdcard.md:15]: URL should be `https://github.com/om2804/octoprint_mkswifisdcard/archive/main.zip`. to match the branch name on GitHub.

**Review by jneilliii — CHANGES_REQUESTED** [review]
One little change to your template, and you're still missing the 1.0.0 tagged release [here](https://github.com/om2804/octoprint_mkswifisdcard/releases).

**Review by jneilliii — APPROVED** [review]

**Review by jneilliii — APPROVED** [review]

**Pull request merged**